src/transformers/hf_logging.py

```diff
@@ -0,0 +1,181 @@
+# limitations under the License.
+""" Logging utilities. """
+
+import logging
+import threading
+from logging import CRITICAL  # NOQA
+from logging import DEBUG  # NOQA
+from logging import ERROR  # NOQA
+from logging import FATAL  # NOQA
+from logging import INFO  # NOQA
+from logging import NOTSET  # NOQA
+from logging import WARN  # NOQA
+from logging import WARNING  # NOQA
+from typing import Optional
+
+
+_lock = threading.Lock()
+_default_handler: Optional[logging.Handler] = None
+
+
+def _get_library_name() -> str:
+
+    return __name__.split(".")[0]
+
+
+def _get_library_root_logger() -> logging.Logger:
+
+    return logging.getLogger(_get_library_name())
+
+
+def _configure_library_root_logger() -> None:
+
+    global _default_handler
+
+    with _lock:
+        if _default_handler:
+            # This library has already configured the library root logger.
+            return
+        _default_handler = logging.StreamHandler()  # Set sys.stderr as stream.
+
+        # Apply our default configuration to the library root logger.
+        library_root_logger = _get_library_root_logger()
+        library_root_logger.addHandler(_default_handler)
+        library_root_logger.setLevel(logging.INFO)
+        library_root_logger.propagate = False
+
+
+def _reset_library_root_logger() -> None:
+
+    global _default_handler
+
+    with _lock:
+        if not _default_handler:
+            return
+
+        library_root_logger = _get_library_root_logger()
+        library_root_logger.removeHandler(_default_handler)
+        library_root_logger.setLevel(logging.NOTSET)
+        _default_handler = None
+
+
+def get_logger(name: Optional[str] = None) -> logging.Logger:
+    """Return a logger with the specified name.
+    This function is not supposed to be directly accessed by library users.
+    """
+
+    if name is None:
+        name = _get_library_name()
+
+    _configure_library_root_logger()
+    return logging.getLogger(name)
+
+
+def get_verbosity() -> int:
+    """Return the current level for the HuggingFace Transformers's root logger.
+    Returns:
+        Logging level, e.g., ``transformers.hf_logging.DEBUG`` and ``transformers.hf_logging.INFO``.
+    .. note::
+        HuggingFace Transformers has following logging levels:
+        - ``transformers.hf_logging.CRITICAL``, ``transformers.hf_logging.FATAL``
+        - ``transformers.hf_logging.ERROR``
+        - ``transformers.hf_logging.WARNING``, ``transformers.hf_logging.WARN``
+        - ``transformers.hf_logging.INFO``
+        - ``transformers.hf_logging.DEBUG``
+    """
+
+    _configure_library_root_logger()
+    return _get_library_root_logger().getEffectiveLevel()
+
+
+def is_verbosity_at_least_info():
+    return get_verbosity() <= INFO
+
+
+def is_verbosity_at_least_warning():
+    return get_verbosity() <= WARNING
+
+
+def is_verbosity_at_least_debug():
+    return get_verbosity() <= DEBUG
+
+
+def is_verbosity_at_least_error():
+    return get_verbosity() <= ERROR
```

Comment:
Ok actually I think these ones will be not very useful (cf the tqdm in `file_utils`). Sorry for proposing them.